backend/account_v2/authentication_service.py

```diff
@@ -295,36 +303,108 @@ def get_organization_by_org_id(self, id: str) -> OrganizationData:
         )
         return organizationData
 
-    def set_default_user(self, username: str, password: str) -> bool:
+    def _set_default_user(self) -> bool:
         """Set the default user for authentication.
 
-        This method creates a default user with the provided username and
-        password if the username and password match the default values defined
-        in the 'DefaultOrg' class. The default user is saved in the database.
+        Creates or updates a default user with predefined credentials.
+
+        Returns:
+            bool: True if the default user is successfully created/updated.
+        """
+        try:
+            UserContext.set_organization_identifier(DefaultOrg.ORGANIZATION_NAME)
+            organization = UserContext.get_organization()
+
+            user = self._get_or_create_user(organization)
+            self._update_user_credentials(user)
+
+            return True
+
+        except Exception as e:
+            logger.error(f"Failed to set default user: {str(e)}")
+            return False
+
+    def _get_or_create_user(self, organization: Organization) -> User:
+        """Get existing user or create a new one based on organization context.
 
         Args:
-            username (str): The username of the default user.
-            password (str): The password of the default user.
+            organization: The organization context
 
         Returns:
-            bool: True if the default user is successfully created and saved,
-                False otherwise.
+            User: The retrieved or created user
         """
-        if (
-            username != DefaultOrg.MOCK_USER
-            or password != DefaultOrg.MOCK_USER_PASSWORD
-        ):
-            return False
+        if not organization:
```

Comment:
@muhammad-ali-e [MINOR] Is this check needed here? Will there be a chance for `organization` to be None, when control reaches service layer?

Reply:
@muhammad-ali-e if the above check is needed please change the mypy type of the argument in this function to indicate that its `Optional`

Reply:
@harini-venkataraman it needed for the firsttime to create organization.
@chandrasekharan-zipstack I added optional type